### ISSUE TITLE
`ng add @ng-bootstrap/ng-bootstrap` : Cannot find module '@schematics/angular/utility/project'

### ISSUE THREAD

**Issue by chenfiona** [angular/angular-cli#18046]
# 🐞 Bug report

### Command (mark with an `x`)

- [ ] new
- [ ] build
- [ ] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [x] add
- [ ] update
- [ ] lint
- [ ] xi18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc

### Is this a regression?
 Yes, the previous version in which this bug was not present was: ....
<pre><code>
Angular CLI: 9.1.8
Node: 12.18.1
Package                      Version
------------------------------------------------------
@angular-devkit/architect    0.901.8
@angular-devkit/core         9.1.8
@angular-devkit/schematics   9.1.8
@schematics/angular          9.1.8
@schematics/update           0.901.8
rxjs                         6.5.4
</code></pre>
### Description

 A clear and concise description of the problem...

https://github.com/ng-bootstrap/ng-bootstrap/issues/3788

## 🔬 Minimal Reproduction

> 1) Install CLI ( latest is 10.0.0)
`npm install -g @angular/cli`

> 2) create a new project
`ng new [project-name] `

> 3) `cd` to the project root, install `bootstrap` and update `angular.json`
`npm install save --bootstrap`

> 4) install `ng-bootstrap`
`ng add @ng-bootstrap/ng-bootstrap`

## 🔥 Exception or Error
<pre><code>
[error] Error: Cannot find module '@schematics/angular/utility/project'
Require stack:
C:\mlng-framework\gis-semantics-health\node_modules\@ng-bootstrap\ng-bootstrap\schematics\ng-add\index.js
C:\mlng-framework\gis-semantics-health\node_modules\@angular-devkit\schematics\tools\export-ref.js
C:\mlng-framework\gis-semantics-health\node_modules\@angular-devkit\schematics\tools\index.js
C:\mlng-framework\gis-semantics-health\node_modules\@angular\cli\utilities\json-schema.js
C:\mlng-framework\gis-semantics-health\node_modules\@angular\cli\models\command-runner.js
C:\mlng-framework\gis-semantics-health\node_modules\@angular\cli\lib\cli\index.js
C:\Users\fiona\AppData\Roaming\npm\node_modules\@angular\cli\lib\init.js
C:\Users\fiona\AppData\Roaming\npm\node_modules\@angular\cli\bin\ng
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:966:15)
    at Function.Module._load (internal/modules/cjs/loader.js:842:27)
    at Module.require (internal/modules/cjs/loader.js:1026:19)
    at require (internal/modules/cjs/helpers.js:72:18)
    at Object.<anonymous> (C:\mlng-framework\gis-semantics-health\node_modules\@ng-bootstrap\ng-bootstrap\schematics\ng-add\index.js:6:19)
    at Module._compile (internal/modules/cjs/loader.js:1138:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1158:10)
    at Module.load (internal/modules/cjs/loader.js:986:32)
    at Function.Module._load (internal/modules/cjs/loader.js:879:14)
    at Module.require (internal/modules/cjs/loader.js:1026:19)
    at require (internal/modules/cjs/helpers.js:72:18)
    at new ExportStringRef (C:\mlng-framework\gis-semantics-health\node_modules\@angular-devkit\schematics\tools\export-ref.js:19:25)
    at NodeModulesEngineHost._resolveReferenceString (C:\mlng-framework\gis-semantics-health\node_modules\@angular-devkit\schematics\tools\node-module-engine-host.js:95:21)
    at NodeModulesEngineHost.createSchematicDescription (C:\mlng-framework\gis-semantics-health\node_modules\@angular-devkit\schematics\tools\file-system-engine-host-base.js:174:34)
    at SchematicEngine.createSchematic (C:\mlng-framework\gis-semantics-health\node_modules\@angular-devkit\schematics\src\engine\engine.js:220:38)
    at CollectionImpl.createSchematic (C:\mlng-framework\gis-semantics-health\node_modules\@angular-devkit\schematics\src\engine\engine.js:70:29)
</code></pre>

## 🌍 Your Environment
<pre><code>
>ng version

     _                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/

Angular CLI: 10.0.0
Node: 12.18.0
OS: win32 x64

Angular: 10.0.0
... animations, cli, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router
Ivy Workspace: Yes

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.1000.0
@angular-devkit/build-angular     0.1000.0
@angular-devkit/build-optimizer   0.1000.0
@angular-devkit/build-webpack     0.1000.0
@angular-devkit/core              10.0.0
@angular-devkit/schematics        10.0.0
@ngtools/webpack                  10.0.0
@schematics/angular               10.0.0
@schematics/update                0.1000.0
rxjs                              6.5.5
typescript                        3.9.5
webpack                           4.43.0

</code></pre>

**Anything else relevant?**
I have to downgrade CLI 10.0.0 to CLI 9.1.8.

**Comment by clydin** [comment]
The `add` functionality for a package is contained within the package itself and is not present within the Angular CLI.  A fix for this issue with `ng add @ng-bootstrap/ng-bootstrap` would need to be made to `@ng-bootstrap/ng-bootstrap` package.

The root cause of this particular error has an open issue with the project and can be tracked here: https://github.com/ng-bootstrap/ng-bootstrap/issues/3714

**Comment by chenfiona** [comment]
@clydin :  
`https://github.com/ng-bootstrap/ng-bootstrap/issues/3714` is not the root cause of the CLI error.  
I have followed `https://github.com/ng-bootstrap/ng-bootstrap#installation` but it didn't fix the issue.  I have used the exact same command `ng add @ng-bootstrap/ng-bootstrap` on Wed June 24th CLI 9.1.8. It worked. 
I install another CLI (the system fetches the lastest 10.0.0) and create Project in another folder on Thu June 25th. Then things fall apart. 
These two projects are all the same. Why is the `Private APIs should not be used for the ng add schematic` related to my issue?
